Add peephole for len(x.size())

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59051 Add peephole for len(x.size())**
* #58932 Add more shape functions for mobilenet
* #58890 Switch symbolic shape registry to operator map'
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet

Differential Revision: [D28727247](https://our.internmc.facebook.com/intern/diff/D28727247)

eellison committed:
Add peephole for len(x.size())

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727247).

Krovatkin commented on test/jit/test_peephole.py:
```diff
@@ -529,3 +529,28 @@ def foo(x: int):
+        FileCheck().check("aten::len").run(foo.graph)
+        inputs = list(foo.graph.inputs())
+        inputs[0].setType(inputs[0].type().with_sizes([None, None]))
+        self.run_pass("peephole", foo.graph)
```
should `peephole` be running shape ops by default?

eellison replied:
yes, it is only the weird pipeline in profiling_executor where we shouldnt

Krovatkin reviewed (approved):
:shipit:

eellison committed:
Update on "Add peephole for len(x.size())"


Differential Revision: [D28727247](https://our.internmc.facebook.com/intern/diff/D28727247)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727247).

eellison committed:
Update on "Add peephole for len(x.size())"


Differential Revision: [D28727247](https://our.internmc.facebook.com/intern/diff/D28727247)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727247).
